src/context/tooltipStateSlice.ts

```diff
@@ -0,0 +1,76 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { merge } from 'lodash';
+import { RootState } from './store';
+
+export interface PopupData {
+  [key: string]: { data: number; coordinates: GeoJSON.Position };
+}
+
+export interface MapTooltipState {
+  coordinates?: GeoJSON.Position;
+  locationName: string;
+  data: PopupData;
+  showing: boolean;
+}
+
+const initialState: MapTooltipState = {
+  coordinates: undefined,
```

Comment:
Not a big deal, but if this is literally `undefined`, you don't have to include it. Typescript will yell at you if you forget to include a required initial value in here.